common/src/lazy_pages.rs

```diff
@@ -88,7 +87,21 @@ pub fn protect_pages_and_init_info(
 
     gear_ri::reset_lazy_pages_info();
 
-    gear_ri::set_wasm_mem_begin_addr(mem.get_buffer_host_addr());
+    let addr = match mem.get_buffer_host_addr() {
+        None => {
+            return if !lazy_pages.is_empty() {
+                // In this case wasm buffer cannot be undefined
+                Err(Error::WasmMemBufferIsUndefined)
+            } else {
+                Ok(())
+            };
```

Comment:
```suggestion
            let pages_are_empty = lazy_pages.is_empty();
            // if pages aren't empty, then buffer can't be defined
            return pages_are_empty.then(|| ()).ok_or(Error::WasmMemBufferIsUndefined);
           };
```

Reply:
I think this suggestion is much less understandable